.github/workflows/yocto-build-deploy.yml

```diff
@@ -977,7 +982,14 @@ jobs:
     # we should consider using GitHub hosted runners for the testbot workers.
     runs-on: ${{ matrix.runs_on || fromJSON('["self-hosted", "X64", "kvm"]') }}
     environment: ${{ matrix.environment }}
-    if: ${{ github.event_name != 'push'}}
+    # https://docs.github.com/en/actions/learn-github-actions/expressions#functions
+    # this expression checks to make sure at least one test suite was provided via either matrix syntax
+    if: |
+      github.event_name != 'push' &&
+      (
+        join(fromJSON(inputs.test_matrix).test_suite) != '' ||
+        join(fromJSON(inputs.test_matrix).include.*.test_suite) != ''
+      )
```

Comment:
IIUC, if we define a test matrix including `test_suite` but missing `environment`, the workflow would still break (because the test environment is required but not checked for).

Reply:
Correct, we don't fully prevent shooting ourselves in the foot but we at least do a small sanity of the input.

Reply:
Sure, fair. The extra complexity of checking every corner case probably costs more than it is worth. I am fine with that!